### PR TITLE
Basic Block/LatestVote Caching in DB Refactor

### PR DESCRIPTION
**Pull request by rauljordan**
Follow-up to #3248 

---

# Description

**Write why you are making the changes in this pull request**

This PR adds basic caching for block and validator latest vote getters/putters in our refactored DB.

### CODE REVIEW

File: beacon-chain/db/kv/blocks.go [hunk 1]
```diff
@@ -153,12 +153,9 @@ func (k *Store) BlockRoots(ctx context.Context, f *filters.QueryFilter) ([][]byt
 func (k *Store) HasBlock(ctx context.Context, blockRoot [32]byte) bool {
 	ctx, span := trace.StartSpan(ctx, "BeaconDB.HasBlock")
 	defer span.End()
-	k.blocksLock.RLock()
-	if blk, exists := k.blocks[blockRoot]; exists && blk != nil {
-		k.blocksLock.RUnlock()
+	if v := blockCache.Get(string(blockRoot[:])); v != nil {
```
Comment (nisdas) [inline beacon-chain/db/kv/blocks.go]: why does the cache have keys as strings ? Anything wrong with bytes

Comment (rauljordan) [reply beacon-chain/db/kv/blocks.go]: The cache API requires string keys unfortunately

Comment (nisdas) [reply beacon-chain/db/kv/blocks.go]: maybe we can look at it in another PR

File: beacon-chain/db/kv/blocks.go [hunk 2]
```diff
@@ -4,29 +4,29 @@ import (
...
 	"github.com/pkg/errors"
 	"github.com/prysmaticlabs/go-ssz"
 	"github.com/prysmaticlabs/prysm/beacon-chain/db/filters"
 	ethpb "github.com/prysmaticlabs/prysm/proto/eth/v1alpha1"
-	"github.com/prysmaticlabs/prysm/shared/bytesutil"
 	"github.com/prysmaticlabs/prysm/shared/sliceutil"
 	"go.opencensus.io/trace"
 )
 
+var blockCache = ccache.New(ccache.Configure())
```
Comment (prestonvanloon) [inline beacon-chain/db/kv/blocks.go]: Is this OK? The default is 5000 objects.

https://github.com/karlseguin/ccache/blob/ec06cd93a07565b373789b0078ba88fe697fddd9/configuration.go#L24

**Comment by rauljordan** [comment]
Discussed offline, 1million latest votes cached is ~50Mb, and caching 4 epochs worth of blocks (256), should be more than enough for runtime.

**Comment by rauljordan** [comment]
PTAL @prestonvanloon

**Pull request merged**